File: golang/2414_longest_continuous_substring/longest_continuous_substring.go
// Package main ...
package main

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func longestContinuousSubstring(s string) int {
	res, cur := 1, 1
	for i := 1; i < len(s); i++ {
		if s[i] == s[i-1]+byte(1) {
			cur++
		} else {
			cur = 1
		}
		res = max(res, cur)
	}
	return res
}

func main() {
	tests := []struct {
		s   string
		ans int
	}{
		{"abacaba", 2},
		{"abcde", 5},
	}

	for index, test := range tests {
		assert.Equal(&testing.T{}, test.ans, longestContinuousSubstring(test.s), index)
	}
}

File: cpp/2414_longest_continuous_substring/longest_continuous_substring.cpp
#include <cassert>
#include <string>
#include <tuple>
#include <vector>

using namespace std;

class Solution {
 public:
  int longestContinuousSubstring(string s) {
    int res = 1;
    int cur = 1;
    for (int i = 1; i < s.size(); i++) {
      if (s[i] == s[i - 1] + 1) {
        cur++;
      } else {
        cur = 1;
      }
      res = max(res, cur);
    }
    return res;
  }
};

int main() {
  vector<tuple<string, int>> tests{
      {"abacaba", 2},
      {"abcde", 5},
  };

  for (auto &[s, ans] : tests) {
    assert(Solution().longestContinuousSubstring(s) == ans);
  }
}